### ISSUE TITLE
Misleading StorageClass in DeclarationInformation

### ISSUE THREAD

**Issue by Heizmann** [ultimate-pa/ultimate#424]
According to the documentation StorageClass defines where the declarations of variables and constants are stored.

The values IMPLEMENTATION and PROC_FUNC do not make sense for variables and constants.
Maybe they are used for procedures. This is however questionable since in Boogie declarations of procedures and implementations are global. Maybe we need a different mechanism for procedures/implementations (e.g., separate enum).

**Comment by danieldietsch** [comment]
Why dont they make sense? Can variables not be defined as in or out params of a procedure, function, or implementation?

**Comment by Heizmann** [comment]
Yes, but there are are already the following enum values for the cases that you mentioned.
		PROC_FUNC_INPARAM,
		PROC_FUNC_OUTPARAM,
		IMPLEMENTATION_INPARAM,
		IMPLEMENTATION_OUTPARAM,
